src/Services/OrderService/OrderService.Application/UseCases/Orders/Commands/DeleteOrder/DeleteOrderCommandHandler.cs

```diff
@@ -0,0 +1,22 @@
+﻿using OrderService.Application.Common.Interfaces.Data;
+using OrderService.Application.UseCases.Orders.Exceptions;
+
+namespace OrderService.Application.UseCases.Orders.Commands.DeleteOrder;
+
+public class DeleteOrderCommandHandler(IUnitOfWork unitOfWork)
+    : ICommandHandler<DeleteOrderCommand, bool>
+{
+
+    public async Task<bool> Handle(DeleteOrderCommand request, CancellationToken cancellationToken)
+    {
+        var order = await unitOfWork.OrderRepository.GetAsync(request.OrderId, cancellationToken);
```

Comment:
here you are retrieving order entity from db then giving it's ID back to delete method, which should retrieve entity again from db, this is wrong, if you retrieve entity then pass it to delete method, not only its id